backend/rest_api_service/main.py

```diff
@@ -61,4 +61,125 @@ async def get_all_participants(response: Response, db: Session = Depends(get_db)
 
     except Exception as e:
         response.status_code = 500
-        return {"message": f"An error occured: {e}"}
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/{event_id}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_event_by_id(event_id, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_event_by_id(event_id=event_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/user/{user_id}/events", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_events_of_user_by_user_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_events_of_user_by_user_id(user_id=user_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/crated_by/{user_id}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_events_by_creator_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_events_by_creator_id(user_id=user_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/{event_id}/participants", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_event_participants(event_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_event_participants(event_id=event_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/user/{user_id}/events/range_of_time/{start, end}", tags=['Event'], status_code=status.HTTP_200_OK)
```

Comment:
`start` i `end` zamiast path parameters powinny być query parameters.

Reply:
jeszcze tutaj jakiś błąd wyskakuje. Tak samo jest jak podam datę w formacie timezone aware, czyli np: `2024-01-12T22:11:31+00:00`

<img width="1070" alt="image" src="https://github.com/koteczki01/time_management_web_app/assets/34169914/8d825d50-210c-43cd-9870-219560e56728">

Reply:
@GrzesS0 nadal aktualne

Reply:
Generalnie, to w bazie danych mamy inny format, więc nie ma się co się dziwić, że nie działa (przykład z bd: 2024-04-15T09:00:00). Czy jest jakiś konkretny powód, by użyć query parameters? (czy w innych miejscach też to trzeba pozamieniać?)

Reply:
W tym konkretnym przypadku tak, bo:
- robienie {start, end} powoduje złe formatowanie urla na który wysyłamy zapytanie. Zobacz zdjęcie
- Takie traktuje się jak np. parametry do funkcji, więc tak, trzeba je umieścić jako query parameters

Reply:
Ale i tak formatowanie daty trzeba zmienić, żeby nie trzeba było podawać strefy czasowej. Potrzebujemy tylko dokładność co do godzin